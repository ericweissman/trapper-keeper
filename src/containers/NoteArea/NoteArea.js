import React, { Component } from 'react'
import NoteCard from '../NoteCard/NoteCard'
import { connect } from 'react-redux'
import PropTypes from 'prop-types';
import Masonry from 'react-masonry-css'
import '../../Main.scss'

export class NoteArea extends Component {



  render() {
    const { items, notes } = this.props

    const breakpointColumnsObj = {
      default: 4,
      1100: 3,
      700: 2,
      500: 1
    };

    const noteCards = notes.map(note => {
      const noteItems = items.filter(item => item.noteID === note.id)

      return <NoteCard {...note} noteItems={noteItems} key={note.id}/>
    }).reverse()

    return (
      <Masonry breakpointCols={breakpointColumnsObj}
        className="note-area-grid"
        columnClassName="note-area-grid_column">
        {noteCards}
      </Masonry>
    )
  }
}

export const mapStateToProps = (state) => ({
  notes: state.notes,
  items: state.items
})

NoteArea.propTypes = {
  items: PropTypes.array,
  notes: PropTypes.array,
}

NoteArea.defaultProps = {
  items: [],
  notes: []
}

export default connect(mapStateToProps)(NoteArea)