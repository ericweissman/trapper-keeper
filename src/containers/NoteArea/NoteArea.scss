.note-area-grid {
  display: flex;
  margin-left: -30px;
  width: auto;
}

.note-area-grid_column {
  padding-left: 30px;
  background-clip: padding-box;
}
 
.note-area-grid_column > div {
  background: grey;
  margin-bottom: 30px;
}