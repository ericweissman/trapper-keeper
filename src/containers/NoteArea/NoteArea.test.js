import React from 'react';
import { NoteArea } from './NoteArea';
import { shallow } from 'enzyme';
import { mapStateToProps } from './NoteArea'

describe('NoteArea', () => {
  let wrapper

  it('should match the snapshot', () => {
    //setup
    const mockItems = [{ description: 'Note one of Title one', noteID: 1, id: 1 }, { description: 'Note two of Title one', noteID: 1, id: 2 }, { description: 'Note three of Title one', noteID: 1, id: 3 },
    { description: 'Note one of Title two', noteID: 2, id: 1 }, { description: 'Note two of Title two', noteID: 2, id: 2 }]
    const mockNotes = [{ id: 1, title: "Title One" }, { id: 2, title: "Title Two" }]
    const isLoading = false
    wrapper = shallow(<NoteArea
      isLoading={isLoading}
      items={mockItems}
      notes={mockNotes}
    />)
    //expectation
    expect(wrapper).toMatchSnapshot()

  })

  it('should match the snapshot if the notes are still loading', () => {
    //setup
    const mockItems = [{ description: 'Note one of Title one', noteID: 1, id: 1 }, { description: 'Note two of Title one', noteID: 1, id: 2 }, { description: 'Note three of Title one', noteID: 1, id: 3 },
    { description: 'Note one of Title two', noteID: 2, id: 1 }, { description: 'Note two of Title two', noteID: 2, id: 2 }]
    const mockNotes = [{ id: 1, title: "Title One" }]
    const isLoading = true
    wrapper = shallow(<NoteArea
      isLoading={isLoading}
      items={mockItems}
      notes={mockNotes}
    />)
    //expectation
    expect(wrapper).toMatchSnapshot()

  })

  describe('mapStateToProps', () => {

    it('should return an object with a notes array and items array', () => {
      //setup
      const mockState = { notes: [{}, {}], items: [{}, {}, {}, {}, {}], favorites: [] }
      const expected = { notes: [{}, {}], items: [{}, {}, {}, {}, {}] }
      //execution
      const mappedProps = mapStateToProps(mockState)
      //expectation
      expect(mappedProps).toEqual(expected)
    })
  })
})