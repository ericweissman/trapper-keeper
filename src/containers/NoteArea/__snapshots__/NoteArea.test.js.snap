// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteArea should match the snapshot 1`] = `
<Masonry
  breakpointCols={
    Object {
      "1100": 3,
      "500": 1,
      "700": 2,
      "default": 4,
    }
  }
  className="note-area-grid"
  columnClassName="note-area-grid_column"
>
  <NoteCard
    id="2"
    key="2"
    noteItems={
      Array [
        Object {
          "description": "Note one of Title two",
          "id": "1",
          "noteID": "2",
        },
        Object {
          "description": "Note two of Title two",
          "id": "2",
          "noteID": "2",
        },
      ]
    }
    title="Title Two"
  />
  <NoteCard
    id="1"
    key="1"
    noteItems={
      Array [
        Object {
          "description": "Note one of Title one",
          "id": "1",
          "noteID": "1",
        },
        Object {
          "description": "Note two of Title one",
          "id": "2",
          "noteID": "1",
        },
        Object {
          "description": "Note three of Title one",
          "id": "3",
          "noteID": "1",
        },
      ]
    }
    title="Title One"
  />
</Masonry>
`;

exports[`NoteArea should match the snapshot if the notes are still loading 1`] = `<Loading />`;

exports[`NoteArea should match the snapshot if the notes loaded but there are none to display 1`] = `<NoCards />`;
